helix-tui/src/widgets/table.rs

```diff
@@ -457,27 +464,32 @@ impl<'a> Table<'a> {
                         width: *width,
                         height: table_row.height,
                     },
+                    truncate,
                 );
                 col += *width + self.column_spacing;
             }
         }
     }
 }
 
-fn render_cell(buf: &mut Buffer, cell: &Cell, area: Rect) {
+fn render_cell(buf: &mut Buffer, cell: &Cell, area: Rect, truncate: bool) {
     buf.set_style(area, cell.style);
     for (i, spans) in cell.content.lines.iter().enumerate() {
         if i as u16 >= area.height {
             break;
         }
-        buf.set_spans(area.x, area.y + i as u16, spans, area.width);
+        if cell.content.width() > 1 && truncate {
```

Comment:
I dont think the emptyness check is required here. Why did you add it?

In any case it should be using byte Len (or `is_empty` really) instead of grapheme width since it's O(1) and any non-empty string usually has a width (the cases where it doesn't can be negelagted IMO)